MultiDeviceTestContext example

DrewDevereux opened the pull request:
This pull request provides an example of using MultiDeviceTestContext. It provides a pytest fixture with a workaround for the name resolution issue decribed in #357.

DrewDevereux committed:
Add example of using MultiDeviceTestContext with name resolution issue workaround

ajoubertza commented:
Thanks @DrewDevereux.  Would you mind changing to be something that is standalone (i.e., runnable out of the box)?  It shouldn't depend on non-standard packages like ska.base and ska.mccs.  It would be enough to add the two trivial devices with the required properties, attributes and commands to `test_integration.py`.

DrewDevereux committed:
Convert to standalone

DrewDevereux commented:
Okay, the `test_integration.py` now contains two simple devices `Master` and `Worker`, and the tests are passing standalone now.

ajoubertza committed:
Restore multidevicetestcontext example docstrings

Useful to know that pytest-mock must be installed.

ajoubertza committed:
Use super() call in multidevicetestcontext example

Just good practice to call this, so include in the example.  Using Python 2 style, since PyTango still supports both Python 2 and 3.

ajoubertza reviewed (approved):
Looks good - thanks for adding this nice example, @DrewDevereux.
